### PR TITLE
[DGS-161] Add a `subsidies-external` feature flag

### PR DESCRIPTION
**Pull request by wolfderechter**
## ID
DGS-161

## Description

Add a `subsidies-external` feature flag to quickly change from internal subsidiemodule in loket to external subsidies application redirection.

In the docker-compose.override.yml for production deployment should the following environment variables be added with the final values (TBD):
```
EMBER_SUBSIDIES_URL: "https://subsidiepunt.lokaalbestuur.vlaanderen.be/"
EMBER_FEATURE_SUBSIDIES_EXTERNAL: "false"
```

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

## How to test

During development, update the `config/environment.js` environment variables of `subsidies-external` to true/false and `subsidiesUrl` to a new url. The subsidy module should redirect to the url when the feature flag is set to true.

### CODE REVIEW

**Review by Windvis — CHANGES_REQUESTED** [review]

**Comment by Windvis** [comment]
Looks good, and worked as expected once I configured the environment variables so should be good to merge once the small issues are resolved.

File: app/routes/subsidies-external-redirect.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import { inject as service } from '@ember/service';
+
+export default class SubsidiesExternalRedirectRoute extends Route {
+  @service router;
+
+  model(params, transition) {
+    if (!config.subsidiesUrl.startsWith('{{')) {
+      const subsidiesUrl = new URL(config.subsidiesUrl);
+      if (transition.to.localName === 'with-path') {
+        const path = '/subsidy/' + transition.to.params.path;
```
[the remark this answers is not in the document]
Comment (wolfderechter) [reply app/routes/subsidies-external-redirect.js:12]: We are currently keeping the default routes, but have plans to indeed remove the `/subsidy` route one level down if everything works and we have some time

File: app/components/shared/compact-menu.hbs [hunk 2]
```diff
@@ -47,7 +47,16 @@
...
   {{/if}}
-  {{#if this.currentSession.canAccessSubsidies}}
+  {{#if (is-feature-enabled "subsidies-external")}}
+      <AuLinkExternal
+        @icon="link-external"
+        @iconAlignment="left"
+        href={{(subsidies-url)}}
+        role="menuitem"
+      >
+        Subsidiebeheer
```
[the remark this answers is not in the document]
Comment (wolfderechter) [reply app/components/shared/compact-menu.hbs:57]: Good idea!

File: app/templates/subsidies-external-redirect.hbs
Comment (Windvis) [inline app/templates/subsidies-external-redirect.hbs:1]: The rename is needed here as well

Comment (wolfderechter) [reply app/templates/subsidies-external-redirect.hbs:1]: ah yes indeed

**Review by Windvis — APPROVED** [review]

**Comment by Windvis** [comment]
It seems we forgot to release this. It's part of the [v0.94.0 release](https://github.com/lblod/frontend-loket/blob/development/CHANGELOG.md#v0940-2024-06-19) now.

**Pull request merged**